src/ethereum/cancun/state.py

```diff
@@ -603,3 +616,90 @@ def get_storage_original(state: State, address: Address, key: Bytes) -> U256:
     assert isinstance(original_value, U256)
 
     return original_value
+
+
+def backup_transient_storage(transient_storage: TransientStorage) -> None:
```

Comment:
Does this function mark where a rollback will rollback to? Like this takes a snapshot? If so, I think this function's name should be something to do with "begin". It's pretty recognized as how you begin a transaction. We do this ourselves up in `begin_transaction`.

Reply:
Updated